### ISSUE TITLE
False positive for F821 in cases of `del` and function scopes

### ISSUE THREAD

**Issue by KennethEnevoldsen** [astral-sh/ruff#20094]
### Summary

I get a F821 when I have a variable called inside a function scope (`test()`), if the scope is followed by `del`.

```py
# tmp.py
for i in range(10):
    a = 2
    print(a) # works fine
    del a

for i in range(10):
    a = 2
    def test():
        print(a) # Undefined name 'a'
    test()
    del a
```

errors are obtained using:
```
ruff check tmp.py
```

### Other

Here is a few more observations that might help locate the bug. Removing the `del` solves the issue
```py
for i in range(10):
    a = 2
    def test():
        print(a) # works fine
    test()
```

The loop is also not required to this to appear:
```py
def test():
    print(a)  # Undefined name 'a'

test()
del a
```
but the function scope is.

### Version

ruff 0.12.9 (ef422460d 2025-08-14)

**Comment by dylwil3** [comment]
Unless I'm misunderstanding this lint is correct. We can't know when and where you will call that function, and whether it's called after you have done `del a`. So, for example:

```pycon
>>> a = 1
>>> def test():
...     print(a)
...
>>> del a
>>> test()
Traceback (most recent call last):
  File "<python-input-3>", line 1, in <module>
    test()
    ~~~~^^
  File "<python-input-1>", line 2, in test
    print(a)
          ^
NameError: name 'a' is not defined
```

**Comment by KennethEnevoldsen** [comment]
Ahh, right, so it's a case that is unknowable - so the lint is correct - thanks for the clarification :)

**Comment by ntBre** [comment]
There's also https://github.com/astral-sh/ruff/issues/9858, which sounds closely related.
